### PR TITLE
JAMES-2915 Test and fix default server/app configuration

### PR DESCRIPTION
**Pull request by chibenwa**
Note: because we should bind a random port in test, protocol related configuration needs to be overriden by tests.

### CODE REVIEW

**Comment by chibenwa** [comment]
server app tests overriden this, docker spring uses its own config... hence no tests

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
`DiscardProtocolTest.testRecord:84 »  Timed out wait for be notified that read ...`

test this please

File: server/app/src/test/java/org/apache/james/app/spring/JamesSpringContextTest.java [hunk 1]
```diff
@@ -50,6 +50,6 @@ public void springShouldLoadAndAddOnlyOneQuotaUpdaterListener() {
         InVMEventBus eventBus = context.getBean(InVMEventBus.class);
 
         assertThat(eventBus.registeredGroups())
-            .containsExactlyInAnyOrder(ListeningCurrentQuotaUpdater.GROUP, LuceneMessageSearchIndex.GROUP);
+            .containsExactlyInAnyOrder(MailboxAnnotationListener.GROUP, LuceneMessageSearchIndex.GROUP);
```
Comment (mbaechler) [inline server/app/src/test/java/org/apache/james/app/spring/JamesSpringContextTest.java]: how is it related?

Comment (chibenwa) [reply server/app/src/test/java/org/apache/james/app/spring/JamesSpringContextTest.java]: This is outdated.

Shortly: main configuration files do have more mailbox listeners than the ones previously used for testing.

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**